File: packages/shared/src/ipc/contracts/github.ts
/**
 * GitHub repos import — fetch a user's public repos for the resume-builder
 * "Import from GitHub" projects step. The backend extracts + validates the
 * username (bare name or `github.com/<user>` URL), drops forks, sorts by stars,
 * and caps to the top 30. Fields are camelCase to match the Rust output struct.
 */

/** A single public GitHub repo offered to the candidate for import. */
export interface GitHubRepo {
  name: string;
  /** Omitted by the backend when absent (serde `skip_serializing_if`). */
  description?: string;
  /** Canonical repo URL — kept verbatim; the AI step never rewrites it. */
  htmlUrl: string;
  language?: string;
  topics: string[];
  /** `stargazers_count` from the GitHub API. */
  stars: number;
  pushedAt?: string;
}

export interface GitHubContract {
  /**
   * Fetch a user's public repos. `input` is a bare username or a
   * `github.com/<user>` URL. Resolves to the repo list (the `{ repos }`
   * envelope is unwrapped in the client layer); rejects on validation /
   * rate-limit / not-found errors.
   */
  importRepos(input: string): Promise<GitHubRepo[]>;
}

export const GITHUB_CHANNELS = {
  importRepos: 'github_import_repos',
} as const;

File: apps/desktop/src-tauri/src/commands/github.rs
use serde_json::{json, Value};

/// Fetch a user's public GitHub repos for the resume-builder import flow.
///
/// `input` is a bare username or a `github.com/<user>` URL; the backend extracts
/// + validates the username and constructs the api.github.com URL itself (the
/// renderer never supplies the request URL). Returns `{ repos: [...] }` on Ok or
/// `{ error: "..." }` on failure — mirrors `profile_import_from_url`'s envelope.
#[tauri::command]
pub async fn github_import_repos(input: String) -> Value {
    match crate::profile_import::github::fetch_repos(&input).await {
        Ok(repos) => json!({ "repos": repos }),
        Err(e) => json!({ "error": e.to_string() }),
    }
}
